cwl/src/main/scala/wdl4s/cwl/WorkflowStep.scala

```diff
@@ -182,8 +220,29 @@ case class WorkflowStepOutput(id: String)
 
 object WorkflowStep {
 
+  private [cwl] object WorkflowStepInputFold {
+    private [cwl] def emptyRight = WorkflowStepInputFold(Map.empty, Set.empty).asRight[NonEmptyList[String]]
+  }
+  private [cwl] case class WorkflowStepInputFold(stepInputMapping: Map[String, ExpressionNode], generatedNodes: Set[GraphNode]) {
+    def withMapping(stepInput: String, value: ExpressionNode) = this.copy(stepInputMapping = stepInputMapping + (stepInput -> value))
+    def withNewNodes(newNodes: Set[GraphNode]) = this.copy(generatedNodes = generatedNodes ++ newNodes)
+    def withNewNode(newNode: GraphNode) = this.copy(generatedNodes = generatedNodes + newNode)
+  }
+
   val emptyOutputs: Outputs = Coproduct[Outputs](Array.empty[String])
 
+  implicit class EnhancedWorkflowStepInput(val workflowStepInput: WorkflowStepInput) extends AnyVal {
+    def toExpressionNode(sourceMappings: Map[String, OutputPort]): ErrorOr[ExpressionNode] = {
+      val womExpression = PlaceholderWomExpression(sourceMappings.keySet, WdlAnyType)
+      ExpressionNode.linkWithInputs(workflowStepInput.id, womExpression, sourceMappings)
+    }
+  }
+
+  object InputSourcesFold extends Poly1 {
+    implicit def one: Case.Aux[String, Set[String]] = at[String] { Set(_): Set[String] }
+    implicit def many: Case.Aux[Array[String], Set[String]] = at[Array[String]] { _.toSet: Set[String] }
```

Comment:
You can dispense with the ascriptions inside the curlies now that you have the `Case.Aux` ascriptions